lib/actuator/include/actuator/LiftsLimitSwitchesManager.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) CONFIG_ACTUATOR_NUMBER25 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+/// @ingroup     actuator
+/// @{
+/// @file
+/// @brief       Singleton managing GPIO-based limit switches for motors.
+/// @author      Gilles DOFFE <g.doffe@gmail.com>
+
+#pragma once
+
+// RIOT includes
+#include <event.h>
+#include <periph/gpio.h>
+
+// Project includes
+#include "actuator/Lift.hpp"
+
+// Packages includes
+#include "etl/map.h"
+#include "etl/delegate.h"
+#include "etl/pool.h"
+
+
+namespace cogip {
+namespace actuators {
+namespace positional_actuators {
+
+/// @brief Singleton managing GPIO-based limit switches for motors.
+/// @details This class handles GPIO interrupts via a dedicated thread,
+///          allowing motor control to react to events such as limit switches
+///          in a safe, non-blocking, and asynchronous way.
+class LiftsLimitSwitchesManager {
+public:
+    /// @brief Callback type used to react to GPIO events in thread context.
+    using GpioCallback = etl::delegate<void(void)>;
```

Comment:
I didn't see any uses of `GpioCallback` ? Maybe I miss something or this line should be removed.

Reply:
Oooops oversight ! Done.